app/lib/whitelabel_detection_middleware.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class WhitelabelDetectionMiddleware
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    Whitelabel.reset!
+    request = ActionDispatch::Request.new(env)
+    Whitelabel.label = detect_label(request)
+    @app.call(env)
+  end
+
+  private
+
+  def detect_label(request)
+    Whitelabel.label_for(request.subdomains.first).tap { return _1 unless _1.nil? }
```

Comment:
is this the new anonymous it block syntax? i'm not used to that.

Reply:
> Implicit Block Arguments
> 
> The thing you can do in blocks that you can’t do in methods is implicitly refer to positional arguments in blocks without naming them. The first argument is _1, the second is _2 and so on. You should probably only use this feature with one argument…
> 
> [1, 2, 3].map { _1 * _1 }
> 
> In Ruby 3.4, it will be available as a synonym for _1, so you will be able to write this code as:
> 
> [1, 2, 3].map { it * it }
> 
> I kind of like it?

hm, i've never heard of positional block arguments in rails. not a fan though. one of the things that i hated most about the groovy scripting language were the "anonymous" implicit block variables that came from nowhere. i think it's worth putting in the additional 4 characters to make it easy to follow.

Reply:
newer ruby version also support https://bugs.ruby-lang.org/issues/18980

Reply:
done